pkg/util/chanmanager/chanmanager.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Channel Manager keeps track of multiple channels
+package chanmanager
+
+import (
+	"fmt"
+	"sync"
+)
+
+type ChanManager interface {
+	// Creates a new channel with specified buffer size. Adds it to manager with the specified chanID.
+	// Returns a read-only version of the newly created channel.
```

Comment:
I was debating what to do in that case. I decided I didn't really care, since my code handled the cleanup. But to make this more generally useful, it would be good to error if an entry already exists. Done.